vars/daosLatestVersion.groovy

```diff
@@ -23,13 +23,12 @@ String call(String next_version='1000', String repo_type='stable') {
                 case 'weekly-testing':
                     _next_version = 1000
                     break
-                case 'release/2.2':
-                case 'weekly-testing-2.2':
-                    _next_version = 2.3
-                    break
-                case 'release/2.0':
-                case 'weekly-testing-2.0':
-                    _next_version = 2.1
+                case ~/^(release|weekly).*(\/|-)(\d+\.\d+)/:
+                    version = Matcher.lastMatcher.group(3)
+                    major = version.split('.')[0] as Integer
+                    minor = version.split('.')[1] as Integer
+                    minor++
+                    _next_version = new BigDecimal("${major}.${minor}")
```

Comment:
How does this all know when the _next version_ is actually a major version bump and not just a minor?

Reply:
I suspect when we create `release/3.0` the `next_version` for the last `release/2.x` will still be `2.(x+1)` and it'll be `3.1` for the latest release branch.